commit 623e95394e0e62e7f9ced2104b786d21e9c0bf53
Author: Đoàn Trần Công Danh <congdanhqx@gmail.com>
Date:   Wed Sep 21 22:21:58 2022 +0700

    cpython: fix build with Python 3.11
    
    Python 3.11 moves longintrepr.h into cpython sub-directory.
    However, in this version, longintrepr.h is always included.

Index: collectd/src/cpython.h
===================================================================
--- collectd.orig/src/cpython.h
+++ collectd/src/cpython.h
@@ -24,9 +24,15 @@
  *   Sven Trenkel <collectd at semidefinite.de>
  **/
 
+#include <Python.h>
 /* Some python versions don't include this by default. */
-
+#if PY_VERSION_HEX < 0x030B0000
+/*
+ * Python 3.11 move longintrepr.h to cpython/longintrepr.h
+ * And it's always included
+ */
 #include <longintrepr.h>
+#endif /* PY_VERSION_HEX < 0x030B0000 */
 
 /* These two macros are basically Py_BEGIN_ALLOW_THREADS and
  * Py_BEGIN_ALLOW_THREADS
